File: Data Structures & Algorithms/string-encode-and-decode/submission-6.py
class Solution:

    def encode(self, strs: List[str]) -> str:
        parts = []
        for s in strs:
            tmp = str(len(s)) + '#' + s
            parts.append(tmp)
        return "".join(parts)

    def decode(self, s: str) -> List[str]:
        i = 0
        ans = []
        while i<len(s):
            j = i
            while s[j] != '#':
                j += 1
            x = int(s[i:j])
            j += 1
            ans.append(s[j:j+x])
            i = j+x
        return ans

File: Data Structures & Algorithms/climbing-stairs/submission-6.java
class Solution {
    public int climbStairs(int n) {
        if(n==1) return 1;
        int x=1, y=2;
        for(int i=3; i<=n; i++) {
            int tmp = x+y;
            x=y;
            y=tmp;
        }
        return y;
    }
}
